src/App.tsx

```diff
@@ -1,61 +1,108 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isGoodToSubmit, setIsGoodToSubmit] = useState(false);
+  const [selectedTodos, setSelectedTodos] = useState(todos);
+
+  const setNewTodoId = (prevTodos: Todo[]) => {
+    const findLastId = Math.max(...prevTodos.map(todo => todo.id));
+
+    return findLastId + 1;
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId || !title.trim()) {
+      setIsGoodToSubmit(true);
+    } else {
+      const addedTodo = {
+        id: setNewTodoId(selectedTodos),
+        title,
+        completed: false,
+        userId,
+        user: getUserById(userId),
+      };
+
+      setSelectedTodos(curTodos => ([
+        ...curTodos,
+        addedTodo,
+      ]));
+
+      setUserId(0);
+      setTitle('');
+      setIsGoodToSubmit(false);
```

Comment:
maybe it would be easier to manage if you created 2 states for title error and user error and check them on input change and form submit?